include/mbedtls/ssl.h

```diff
@@ -382,6 +388,28 @@ union mbedtls_ssl_premaster_secret
 
 #define MBEDTLS_PREMASTER_SIZE     sizeof( union mbedtls_ssl_premaster_secret )
 
+static inline int mbedtls_ssl_chk_buf( uint8_t *cur, const uint8_t *end, size_t need )
+{
+    return( end < cur || (size_t)( end - cur ) < (size_t) need );
+}
+
+#define MBEDTLS_SSL_CHK_BUF_PTR( cur, end, need )			\
+    do {								\
+        if( mbedtls_ssl_chk_buf( ( cur ), ( end ), ( need ) ) != 0 )	\
+        {								\
+            return( MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL );			\
+        }								\
+    } while( 0 )
```

Comment:
@andresag01 The `MBEDTLS_SSL_CHK_BUF_PTR_WITH_DEBUG` is used solely in extension writing routines or in places where there was no debug output previously. As the extension writing functions have return values now, and `MBEDTLS_SSL_CHK_BUF` throws a descriptive error `MBEDTLS_SSL_ERR_BUFFER_TOO_SMALL` on failure, it might be acceptable to drop the additional debugging output in these places, thereby removing the need for `MBEDTLS_SSL_CHK_BUF_PTR_WITH_DEBUG` altogether. Regarding the location of `MBEDTLS_SSL_CHK_BUF_PTR` and `mbedtls_ssl_chk_buf`, the options seem to be to duplicate or to move it to `ssl_internal.h`, but not leave it in `ssl.h`. As far as I understood @yanesca, we don't treat the contents of `ssl_internal.h` as part of the public API, so moving it there should be acceptable?